mixnet/protocol/src/lib.rs

```diff
@@ -55,6 +88,51 @@ impl Body {
         Self::sphinx_packet_from_bytes(&buf)
     }
 
+    async fn read_ack_response<R>(
+        reader: &mut R,
+    ) -> Result<Body, Box<dyn Error + Send + Sync + 'static>>
+    where
+        R: AsyncRead + Unpin,
+    {
+        let cks = reader.read_u32().await?;
+        let id = PacketId(cks);
+        let socket_type = reader.read_u8().await?;
+        match socket_type {
+            0 => {
+                let mut buf = [0; 6];
+                reader.read_exact(&mut buf).await?;
+                let id = PacketId::from_bytes(&buf);
+                let ip_addr = Ipv4Addr::new(buf[0], buf[1], buf[2], buf[3]);
+                let port = u16::from_be_bytes([buf[4], buf[5]]);
+                Ok(Self::AckResponse(AckResponse {
+                    id,
+                    sender: SocketAddr::V4(SocketAddrV4::new(ip_addr, port)),
+                }))
+            }
+            1 => {
+                let mut buf = [0; 18];
+                reader.read_exact(&mut buf).await?;
+                // Manually inline here for better performance?
+                let ip_addr = Ipv6Addr::new(
+                    u16::from_be_bytes([buf[0], buf[1]]),
+                    u16::from_be_bytes([buf[2], buf[3]]),
+                    u16::from_be_bytes([buf[4], buf[5]]),
+                    u16::from_be_bytes([buf[6], buf[7]]),
+                    u16::from_be_bytes([buf[8], buf[9]]),
+                    u16::from_be_bytes([buf[10], buf[11]]),
+                    u16::from_be_bytes([buf[12], buf[13]]),
+                    u16::from_be_bytes([buf[14], buf[15]]),
+                );
```

Comment:
Can't this read from the slice directly? I do not think that would be a bottlenetck 🤔